final/scripts/app.js

```diff
@@ -149,37 +167,64 @@
    *
    ****************************************************************************/
 
-  // Gets a forecast for a specific city and update the card with the data
+  /*
+   * Gets a forecast for a specific city and updates the card with the data.
+   * getForecast() first checks if the weather data is in the cache. If so,
+   * then it gets that data and populates the card with the cached data.
+   * Then, getForecast() goes to the network for fresh data. If the network
+   * request goes through, then the card gets updated a second time with the
+   * freshest data.
+   */
   app.getForecast = function(key, label) {
-    var url = 'https://publicdata-weather.firebaseio.com/';
-    url += key + '.json';
+    /*
+     * IDs map to Yahoo Weather API Where On Earth Identifiers (WOEIDs).
+     * https://developer.yahoo.com/weather/documentation.html#req
+     */
+    var ids = {
+      "austin": 2357536,
+      "boston": 2367105,
+      "chicago": 2379574,
+      "newyork": 2459115,
+      "portland": 2475687,
+      "sanfrancisco": 2487956,
+      "seattle": 2490383
+    };
+    var statement = 'select * from weather.forecast where woeid=' + ids[key];
+    var url = 'https://query.yahooapis.com/v1/public/yql?format=json&q=' +
+        statement;
     if ('caches' in window) {
+      /*
+       * Check if the service worker has already cached this city's weather
+       * data. If the service worker has has the data, then display the cached
+       * data while the app fetches the latest data.
+       */
       caches.match(url).then(function(response) {
         if (response) {
-          response.json().then(function(json) {
+          response.json().then(function updateFromCache(json) {
```

Comment:
It helps with debugging stack traces. Instead of `anonymous function` you see `updateFromCache`. 

https://github.com/getify/You-Dont-Know-JS/blob/master/scope%20%26%20closures/ch3.md#anonymous-vs-named

But I'm not doing this consistently throughout the code so I think I'll remove it.